portfolio/src/main/webapp/script.js

```diff
@@ -28,4 +28,33 @@ function addRandomFact() {
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
-}
\ No newline at end of file
+}
+
+/**
+ * Fades in / out div section once the window reaches the top-quarter of the div.
+ * Works with individual or multiple divs.
+ */
+function fadeDiv(classOrIdName) {
+    var window_bottom = $(window).scrollTop() + $(window).height();
+
+    // Conditionally fade each div of specified class or id name.
+    $(classOrIdName).each(function() {
+        var object_top_quarter = $(this).offset().top + $(this).outerHeight() / 4;
+
+        // Fade div if window reached top-quarter marker.
+        if (window_bottom > object_top_quarter) {
+            $(this).addClass('fade-in');
+        } else {
+            $(this).removeClass('fade-in');
+        }
+    });
+}
+
+// Triggered upon DOM load.
+$(document).ready(() => {
+    fadeDiv('.project');
+    // Event every time user scrolls.
+    $(window).scroll(() => {
+        fadeDiv('.project');
+    });
```

Comment:
this is an edge case, but it may be worth adding a trigger to call fadeDiv when the window is resized, since that can change the window height value you get above. Eg: if a user changed the orientation of their phone, or resized their browser while viewing the page.

Reply:
Ah, that's a good point -- thanks for the note, I just added that!